### DISCUSSION TITLE
Volume control based on sensor value

### DISCUSSION THREAD

**Discussion by rav345** [pschatzmann/arduino-audio-tools#863] [Q&A] [answered]
Hi,
I am trying to use an ESP32 Wroom 32 as an A2DPSink and control the volume based on the angular deviations from set angles, using an MPU6050 accelerometer. The breadboard image is below. The code is attached.
The problem: I only hear some "chuck chuck" noise on every iteration of the loop. The volume control does not work either. The sound file attached below.
The example codes for "simple receiver" from the ESP32-A2DP library work fine. 
Please suggest :-)

![IMG_3078 copy](https://github.com/pschatzmann/arduino-audio-tools/assets/37502483/d894e8b1-2a4c-4c8f-b5eb-1248f9677d59)
[MRC3test.docx](https://github.com/pschatzmann/arduino-audio-tools/files/11528877/MRC3test.docx)

https://github.com/pschatzmann/arduino-audio-tools/assets/37502483/3ba05547-1bb0-41a2-af18-ffe054200417

**Comment by pschatzmann** [comment, marked as answer]
Difficult to say what's wrong, but I guess you are just starving the other processes by what you do in your loop.
Try not to call the loop so often e.g. by adding some delay.
